external_parser/example_joiner.cc

```diff
@@ -304,8 +309,19 @@ int example_joiner::process_interaction(const v2::Event &event,
     data.probabilityOfDrop = 1.f - metadata.pass_probability();
     data.skipLearn = cb->deferred_action();
 
-    std::string line_vec(reinterpret_cast<char const *>(cb->context()->data()),
-                         cb->context()->size());
+    const char* context = reinterpret_cast<char const *>(cb->context()->data());
+    joined_event je = {
+      "joiner_timestamp",
+      std::move(meta),
+      std::move(data)
+    };
+
+    if (_binary_to_json) {
+      je.context = context;
+      je.model_id = cb->model_id()->data();
+    }
+
+    std::string line_vec(context, cb->context()->size());
```

Comment:
could we instead of keeping a `const char *` in `joined_event` for the context, keep a `std::string` and copy it into the `je` struct after line 324? I think it's best to copy the string otherwise not sure what happens if the json parser performs destructive parsing